src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM_NUMBER_BOUND = 99;
+    private static final int RANDOM_INDEX = 5;
+    private static final double DEFAULT_RADIUS = 10;
+    private final ColorSupplier colorSupplier;
+    private final Random random;
+
+    public FigureSupplier() {
+        colorSupplier = new ColorSupplier();
+        random = new Random();
+    }
+
+    public Figure getRandomFigure() {
+        Figure square = new Square(random.nextInt(RANDOM_NUMBER_BOUND),
+                        colorSupplier.getRandomColor());
+        Figure rightTriangle = new RightTriangle(random.nextInt(RANDOM_NUMBER_BOUND),
+                        random.nextInt(RANDOM_NUMBER_BOUND), colorSupplier.getRandomColor());
+        Figure rectangle = new Rectangle(random.nextInt(RANDOM_NUMBER_BOUND),
+                        random.nextInt(RANDOM_NUMBER_BOUND), colorSupplier.getRandomColor());
+        Figure circle = new Circle(random.nextInt(RANDOM_NUMBER_BOUND),
+                        colorSupplier.getRandomColor());
+        Figure isoscelesTrapezoid = new IsoscelesTrapezoid(random.nextInt(RANDOM_NUMBER_BOUND),
+                        random.nextInt(RANDOM_NUMBER_BOUND), random.nextInt(RANDOM_NUMBER_BOUND),
+                        colorSupplier.getRandomColor());
+        int randomIndex = random.nextInt(RANDOM_INDEX);
```

Comment:
This method can be optimized. Let's create objects only when we need them 
You can use private methods to make this one short and readable ;)

Reply:
Done